### PR TITLE
Wraps image path update in config option

### PR DESCRIPTION
**Pull request by jonbullock**
Relates to #535 and #459

### CODE REVIEW

File: jbake-core/src/main/java/org/jbake/app/configuration/JBakeConfiguration.java [hunk 1]
```diff
@@ -283,10 +283,15 @@
     boolean getUriWithoutExtension();
 
     /**
-     * @return Flag indicating if image paths should be prepended with {@link #getSiteHost()} value
+     * @return Flag indicating if image paths should be prepended with {@link #getSiteHost()} value - see {@link #getImgPathUpdate()}
```
Comment (kwin) [inline jbake-core/src/main/java/org/jbake/app/configuration/JBakeConfiguration.java]: For me this description is not clear enough. What different effects do I get with the different combinations of boolean values for `img.path.prepend.host` and `img.path.update`?

Comment (kwin) [reply jbake-core/src/main/java/org/jbake/app/configuration/JBakeConfiguration.java]: I don't see `context.rootpath` in https://github.com/jbake-org/jbake/blob/a108a6fb594fac567cbba3a3fd870ca9b1627c08/jbake-core/src/main/java/org/jbake/app/configuration/JBakeProperty.java nor an according getter in https://github.com/jbake-org/jbake/blob/a108a6fb594fac567cbba3a3fd870ca9b1627c08/jbake-core/src/main/java/org/jbake/app/configuration/JBakeConfiguration.java. But if that is the case, I think it is worth mentioning both properties in the javadoc (i.e. site.host and context.rootpath)

Comment (jonbullock) [reply jbake-core/src/main/java/org/jbake/app/configuration/JBakeConfiguration.java]: The `rootpath` variable is calculated for each source content file, which is why it isn't in the config classes, however that variable isn't used in this implementation anymore. 

I'll update the javadoc though to make it clearer though. Thanks @kwin and @manikmagar 👍

**Comment by jonbullock** [comment]
Anyone else have any issue with this being merged in?

**Pull request merged**